tools/downloader/common.py

```diff
@@ -454,18 +454,19 @@ def apply(self, reporter, output_dir):
 Postproc.types['unpack_archive'] = PostprocUnpackArchive
 
 class Model:
-    def __init__(self, name, subdirectory, files, postprocessing, mo_args, quantizable, framework,
+    def __init__(self, name, subdirectory, files, postprocessing, mo_args, quantized, quantizable, framework,
                  description, license_url, precisions, task_type, conversion_to_onnx_args):
         self.name = name
         self.subdirectory = subdirectory
         self.files = files
         self.postprocessing = postprocessing
         self.mo_args = mo_args
+        self.quantized = quantized
         self.quantizable = quantizable
         self.framework = framework
         self.description = description
         self.license_url = license_url
-        self.precisions = precisions
+        self.precisions = {'FP32-INT8', 'FP16-INT8'} if self.quantized else precisions
```

Comment:
This logic should be in `deserialize`, not here. Also, I would suggest to make it more flexible by using, e.g. `precisions = {f'{precision}-{quantized}' for precision in precisions}`.

Reply:
logic moved to deserialize. did not get how to make it more flexible (getting error: local variable 'precisions' referenced before assignment)